main.py

```diff
@@ -31,11 +38,194 @@ def setup(self):
+
+        found_napps = set()
+        content = self.build_content(pipeline)
+        enable_napps = self.get_enabled_napps()
+        # Find NApps in the pipeline to notify them
+        for napp in content:
+            if napp in enable_napps:
+                found_napps.add(napp)
+        self.required_napps = found_napps
+        self.start_enabling_pipeline(content)
+
+    def get_enabled_napps(self):
+        """Get the NApps that are enabled and subscribed"""
+        enable_napps = set()
+        for key in self.controller.napps:
+            # Keys look like this: ('kytos', 'of_lldp')
+            if key[1] in self.subscribed_napps:
+                enable_napps.add(key[1])
+        return enable_napps
+
+    def start_enabling_pipeline(self, content: dict):
+        """Method to start the process to enable table
+        First, send event notifying NApps about their
+        new table set up.
+        """
+        name = "enable_table"
+        self.emit_event(name, content)
+
+    def build_content(self, pipeline: dict):
+        """Build content to be sent through an event"""
+        content = {}
+        with self._pipeline_lock:
+            for table in pipeline["multi_table"]:
+                table_id = table["table_id"]
+                for napp in table["napps_table_groups"]:
+                    if napp not in content:
+                        content[napp] = {}
+                    for flow_type in table["napps_table_groups"][napp]:
+                        content[napp][flow_type] = table_id
+        return content
+
+    def emit_event(self, name: str, content: dict = None):
+        """Send event"""
+        context = "kytos/of_multi_table"
+        event_name = f"{context}.{name}"
+        event = KytosEvent(name=event_name, content=content)
+        self.controller.buffers.app.put(event)
+
+    @listen_to("kytos/(mef_eline|coloring|of_lldp).enable_table")
+    def on_enable_table(self, event):
+        """Listen for NApps responses"""
+        self.handle_enable_table(event)
+
+    def handle_enable_table(self, event):
+        """Handle NApps responses from enable_table
+        Second, wait for all the napps to respond"""
+        napp = event.name.split('/')[1].split('.')[0]
+        # Check against the last current table
+        self.required_napps.remove(napp)
+        if self.required_napps:
+            # There are more required napps, 'waiting' responses
+            return
+        self.get_flows_to_be_installed()
+
+    def get_flows_to_be_installed(self):
+        """Get flows from flow manager so this NApp can modify them
+        Third, install the flows with different table_id"""
+        pipeline = self.get_enabled_table()
+        if pipeline.get("status") == "enabled":
+            # Pipeline enabling has finnished already
+            return
+        # Keys necessary to delete a flow, 'cookie_mask' is added later
+        keys = {'idle_timeout', 'hard_timeout', 'cookie', 'priority',
```

Comment:
For flows deletion `flow_manager` sends a non-strict FlowMod, so you can further simplify here, and only send `cookie`, `cookie_mask` and `match`.

Reply:
Reduced.